cmd/launcher/desktop.go

```diff
@@ -174,20 +170,22 @@ func listenSignals(logger log.Logger) {
 }
 
 // monitorParentProcess continuously checks to see if parent is a live and sends on provided channel if it is not
-func monitorParentProcess(logger log.Logger, ppid int) {
-	ticker := time.NewTicker(2 * time.Second)
+func monitorParentProcess(logger log.Logger, monitorUrl string, interval time.Duration) {
+	ticker := time.NewTicker(interval)
+
+	client := http.Client{
+		Timeout: interval,
+	}
 
 	for ; true; <-ticker.C {
-		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
-		exists, err := process.PidExistsWithContext(ctx, int32(ppid))
-
-		// pretty sure this `cancel()` call is not needed since it should call cancel on it's own when time is exceeded
-		// https://cs.opensource.google/go/go/+/master:src/context/context.go;l=456?q=func%20WithDeadline&ss=go%2Fgo
-		// but the linter and the context.WithTimeout docs say to do it
-		cancel()
-		if err != nil || !exists {
+		response, err := client.Get(monitorUrl)
+		if response != nil && response.Body != nil {
+			response.Body.Close()
+		}
+
+		if err != nil {
```

Comment:
Would we potentially want to wait for maybe ~3 consecutive errors, maybe with a small amount of backoff? Just so that launcher being slow to respond once doesn't immediately kill the desktop process.

Reply:
I added some retries, I started to add a backoff ... but it's fussy to test so I gave up. You think retries is enough?

Reply:
👍 I was worried about something like this too - we could end up with the opposite problem, where desktop/menubar can vanish.